libs/live-config/src/providers/jsonFileReader/index.ts

```diff
@@ -0,0 +1,33 @@
+import fs from "fs";
+import { Provider } from "..";
+import { ConfigInfo } from "../../LiveConfig";
+
+export class JsonFileReader implements Provider {
+  filePath: string;
+
+  constructor(config: { filePath: string }) {
+    if (!config.filePath.endsWith(".json")) {
+      throw new Error("Only .json files can be read by the provider");
+    }
+    this.filePath = config.filePath;
+  }
+
+  getValueBykey<K>(key: K, info: ConfigInfo) {
+    if (!fs.existsSync(this.filePath)) {
```

Comment:
Shouldn't we use an async method ?

Using a sync method here could cause performance issues

Also I wonder if we really want to use fs, isn't it wrong if we port the app in the web ? maybe just a http request for fetching could work

Reply:
This "jsonFileReader" will not be used in production. This is just an example to show how we add a new tailored provider in this lib.

And the firebase provider will be used in production. (In ledger-live/libs/live-config/src/providers/firebaseRemoteConfig/index.ts) There is no performance issues as the getValue won't fetch config from remote server. It just fetch config from local cache. In our current codebase, the "getValue" method is not async either.

For reference, here is the codebase of firebase SDK. As you can see, the design of "getValue" method is not supposed to be async.
https://github.com/firebase/firebase-js-sdk/blob/537d39982f90aff50519b8c1ad6d58048fb8f244/packages/remote-config/src/api.ts#L209

Reply:
If this provider isn't used why don't we just add how to add a provider in the documentation ? I think we should not keep unused code, WDYT ?

Reply:
yes, we can remove it of course. but it is used in the unit tests in live-config lib. This is why we keep it here.
we can't use the firebase provider in unit tests directly as it depends on remove server. so we use this simple provider instead.

Reply:
Will only be used in unit tests